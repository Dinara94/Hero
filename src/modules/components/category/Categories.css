.Categories-container {
    width: 100%;
    margin: 0 auto;
    display: flex;
    flex-flow: column nowrap;
    justify-content: center;
}

.Categories-container__tile {
    width: 80%;
    margin: 50px auto;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-gap: 45px;
}
