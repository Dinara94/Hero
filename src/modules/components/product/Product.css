.Product {
  display: flex;
  flex-flow: column nowrap;
  box-sizing: border-box;
  position: relative;
}

.Product img {
  height: 200px;
  object-fit: cover;
}

.Product__info {
  display: flex;
  flex-flow: column nowrap;
  margin-top: 30px;
  text-align: center;
  color: #7d7d7d;
  font-size: 0.8rem;
}

.Product__name {
  margin: 20px 0;
  color: #000;
  font-size: 0.8rem;
  font-weight: lighter;
  text-transform: uppercase;
  letter-spacing: 1px;
}

.Product__price {
  font-weight: 600;
}

.Product__button {
    display: block;
    position: absolute;
    left: 0;
    right: 0;
    z-index: 1;
    box-sizing: border-box;
    width: 40%;
    margin: 20px auto;
    padding: 10px;
    color: #fff;
    font-size: 0.7rem;
    letter-spacing: ;
    text-align: center;
    text-transform: uppercase;
    cursor: pointer;
}

.details-btn {
    top: 30px;
    background-color: #1d1d20;
}

.add-btn {
    top: 90px;
    background-color: #4a4a4d;
}

.hovered {
    opacity: 0.5;
}

