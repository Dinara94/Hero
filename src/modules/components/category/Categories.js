import React, { useEffect, useState, useCallback } from "react";
import "./Categories.css";
import useProducts from "../../hooks/useProducts.js";
import CategoriesBanner from "./CategoriesBanner.js";
import Product from "../product/Product.js";


const Categories = () => {
  const { list } = useProducts();

  return (
    <section className="Categories-container">
      <CategoriesBanner />
      <div className="Categories-container__tile">
      {list.map((item) => (
          <Product
            key={item.id}
            item={item}
          />
        ))}
      </div>
    </section>
  );
};

export default Categories;
