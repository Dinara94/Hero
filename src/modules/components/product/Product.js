import React, { useState } from "react";
import { Link } from "react-router-dom";
import { useCart } from "../../hooks/CartContext.js";
import PropTypes from 'prop-types';
import "./Product.css";

const Product = ({ item }) => {
  const cart = useCart();
  const [buttonIsHovered, setButtonHovered] = useState(false);

  const addProductTocart = () => {
    cart.sendToCart(item, 1);
  };

  return (
    <figure className="Product">
      <img
        src={`/media/${item.image}`}
        alt={item.title}
        onMouseEnter={() => setButtonHovered(true)}
        onMouseLeave={() => setButtonHovered(false)}
        className={buttonIsHovered ? "hovered" : null}
      />
      {buttonIsHovered && (
        <>
          <Link
            to={`/${item.id}`}
            className="Product__button details-btn"
            onMouseEnter={() => setButtonHovered(true)}
            onMouseLeave={() => setButtonHovered(false)}
          >
            view details
          </Link>
          <button
            className="Product__button add-btn"
            onMouseEnter={() => setButtonHovered(true)}
            onMouseLeave={() => setButtonHovered(false)}
            onClick={addProductTocart}
          >
            add to cart
          </button>
        </>
      )}
      <figcaption className="Product__info">
        <p className="Product__brand">{item.brand}</p>
        <h3 className="Product__name">{item.title}</h3>
        <p className="Product__price">{`\$${item.price.toFixed(2)}`}</p>
      </figcaption>
    </figure>
  );
};

Product.propTypes = {
  item: PropTypes.object,
}

Product.defaultProps = {
  item: {},
}

export default Product;
