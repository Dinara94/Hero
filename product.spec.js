import React from "react";
import Categories from "./src/modules/components/category/Categories.js";
import Product from "./src/modules/components/product/Product.js";
import fileName from "./src/modules/services/services.js";

const example = {
  id: 22386980,
  title: "Hand Painted Blue Flat Dish",
  brand: "Kiriko",
  price: 28,
  description:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
  image: "hand-painted-blue-flat-dish.jpg",
};

const products = [
    {
      id: 22386986,
      title: "Blue Stripe Stoneware Plate",
      brand: "Kiriko",
      price: 40,
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
      image: "blue-stripe-stoneware-plate.jpg"
    },
    {
      id: 22386980,
      title: "Hand Painted Blue Flat Dish",
      brand: "Kiriko",
      price: 28,
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
      image: "hand-painted-blue-flat-dish.jpg"
    },
    {
      id: 22386989,
      title: "Heme",
      brand: "Dust & Form",
      price: 52,
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
      image: "heme.jpg"
    },
    {
      id: 22386981,
      title: "Mashiko-Yaki Green Small Plate",
      brand: "Kiriko",
      price: 28,
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
      image: "mashiko-yaki-green-small-plate.jpg"
    },
    {
      id: 22386903,
      title: "Mashiko-Yaki Indigo Small Plate",
      brand: "Kiriko",
      price: 28,
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
      image: "mashiko-yaki-indigo-small-plate.jpg"
    },
    {
      id: 22386222,
      title: "Mashiko-Yaki Saucer",
      brand: "Kiriko",
      price: 18,
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
      image: "mashiko-yaki-saucer.jpg"
    }
  ]

const mockJsonPromise = Promise.resolve({list: products});
const mockFetchPromise = Promise.resolve({ json: () => mockJsonPromise });
global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

describe("Categories component", () => {
  it("Categories rendering", () => {
    const component = shallow(<Categories />);
    const wrapper = component.find(".Categories-container");
    expect(wrapper.length).toBe(1);
  });

/*   it("should call fetch in useEffect", () => {
    expect(global.fetch).toHaveBeenCalledWith(`${fileName}.json`);
  });
}); */

describe("Product component", () => {
  it("render with props", () => {
    const component = shallow(<Product key={12346} item={example} />);
    expect(component).toMatchSnapshot();
  });

  it("render with default props", () => {
    const result = Product.defaultProps.item;
    expect(result).toStrictEqual({});
  });

  it("should change layout when hovered", () => {
    const component = shallow(<Product key={12346} item={example} />);
    const wrapper = component.find("img");
    wrapper.simulate("mouseenter");
    expect(component).toMatchSnapshot();
  });
});
