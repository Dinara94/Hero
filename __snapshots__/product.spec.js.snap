// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component and children Product component should change layout when hovered 1`] = `
<figure
  className="Product"
>
  <img
    alt="Hand Painted Blue Flat Dish"
    className="hovered"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="/media/hand-painted-blue-flat-dish.jpg"
  />
  <Link
    className="Product__button details-btn"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    to="/22386980"
  >
    view details
  </Link>
  <button
    className="Product__button add-btn"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    add to cart
  </button>
  <figcaption
    className="Product__info"
  >
    <p
      className="Product__brand"
    >
      Kiriko
    </p>
    <h3
      className="Product__name"
    >
      Hand Painted Blue Flat Dish
    </h3>
    <p
      className="Product__price"
    >
      $28.00
    </p>
  </figcaption>
</figure>
`;

exports[`Categories component and children render Product component with props 1`] = `
<figure
  className="Product"
>
  <img
    alt="Hand Painted Blue Flat Dish"
    className={null}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="/media/hand-painted-blue-flat-dish.jpg"
  />
  <figcaption
    className="Product__info"
  >
    <p
      className="Product__brand"
    >
      Kiriko
    </p>
    <h3
      className="Product__name"
    >
      Hand Painted Blue Flat Dish
    </h3>
    <p
      className="Product__price"
    >
      $28.00
    </p>
  </figcaption>
</figure>
`;

exports[`Product component render with props 1`] = `
<figure
  className="Product"
>
  <img
    alt="Hand Painted Blue Flat Dish"
    className={null}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="/media/hand-painted-blue-flat-dish.jpg"
  />
  <figcaption
    className="Product__info"
  >
    <p
      className="Product__brand"
    >
      Kiriko
    </p>
    <h3
      className="Product__name"
    >
      Hand Painted Blue Flat Dish
    </h3>
    <p
      className="Product__price"
    >
      $28.00
    </p>
  </figcaption>
</figure>
`;

exports[`Product component should change layout when hovered 1`] = `
<figure
  className="Product"
>
  <img
    alt="Hand Painted Blue Flat Dish"
    className="hovered"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="/media/hand-painted-blue-flat-dish.jpg"
  />
  <Link
    className="Product__button details-btn"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    to="/22386980"
  >
    view details
  </Link>
  <button
    className="Product__button add-btn"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    add to cart
  </button>
  <figcaption
    className="Product__info"
  >
    <p
      className="Product__brand"
    >
      Kiriko
    </p>
    <h3
      className="Product__name"
    >
      Hand Painted Blue Flat Dish
    </h3>
    <p
      className="Product__price"
    >
      $28.00
    </p>
  </figcaption>
</figure>
`;
